### PR TITLE
Redesign trending card

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T261669

This PR also updates the `header()` and `footer()` methods that use the `getCard()` instead of passing a specific card object.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/feed/view/GraphView.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    private val pathPaint = Paint().apply {
+        style = Paint.Style.STROKE
+        strokeWidth = GRAPH_STROKE_WIDTH
+        strokeCap = Paint.Cap.ROUND
+        pathEffect = CornerPathEffect(GRAPH_STROKE_WIDTH)
+        isAntiAlias = true
+    }
+
+    fun setData(viewHistoryList: List<MostReadArticles.ViewHistory>) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/feed/view/GraphView.kt:27]: It's nice that this class has the generic name `GraphView`, which suggests we'll be able to reuse it for graphing other things.  But then it's using this extremely specific `MostReadArticles.ViewHistory` object for providing the data points. Is there a way for this View class to take a completely generic list of `floats` for graphing, regardless of where the numbers come from?

**Review by dbrant — CHANGES_REQUESTED** [review]
See quick note inline. Also, I'm seeing the circle gradient slightly cut off along the edges:

image

**Comment by cooltey** [comment]
> 
> 
> See quick note inline. Also, I'm seeing the circle gradient slightly cut off along the edges:
> 

Thanks, I guess my previous setting depends on devices. Done.

File: app/src/main/java/org/wikipedia/feed/view/GraphView.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/feed/view/GraphView.kt]: Make sense. Done!

**Pull request merged**